### DISCUSSION TITLE
New feature - Override block size

### DISCUSSION THREAD

**Discussion by Zalorpas** [keirf/greaseweazle#439] [Ideas]
Hello, 

As I mentioned in another post, we are trying to preserve some arcade games that are stored in disks.
One of the issues we found is that the creators were storing blocks of 1024k but they were marking it as blocks of 512k

In order to break the protection, we compiled a homebrew version of HxC to force reading blocks of 1024k overriding the information found in the RAW data (which worked quite well to read the content).

The problem is that, for some games we lost both, the disk and the RAW data, but hopefully, we have the DSK (image).
I am wondering if it is possible to add a parameter in GreaseWeazle to override the block size found by a customized one (for both, reading and writting). 

It will be really appreciated.
Thank you very much,

**Comment by keirf** [comment]
So you want to ignore the sector N field and instead expect the specified sector data size (with valid crc on the end)?

**Comment by Zalorpas** [comment]
Let me detail how it should work.

For reading:
The track has sectors of 1024 but the header is marked as 512 and the CRC was calculated according to this block size.
Without the override, you only see the first 512b and the second half of the information is ignored...
![Format (original 512)](https://github.com/keirf/greaseweazle/assets/170749544/f4381ff2-17ce-4d11-b468-ec6a2327eb53)

In order to get all the information, the header must be ignored, and the block size must be forced to 1024
![Format (forcing 1024)](https://github.com/keirf/greaseweazle/assets/170749544/b6a379e4-07b4-4b6c-bff3-b5414265b1fd)

So, ideally, with a "force-blocksize=1024" parameter for reading, it would allow us to get all the information,

For writing (just the other way round)
We must write sectors of 1024 but those must be marked as 512 and the CRC must be calculated according to this block size.
In addition, the sector ID must start at 200 but I think it would be feasible by editing the diskdefs.cfg file (id=200) and tweaking the parameters should work, but I never tested that before.

If you need any other details or snapshots, please let me know.

Thank you very much for your support

**Reply by keirf** [reply]
Do you mean that the CRC is calculated over 512 bytes only but placed after the 1024th data byte?

**Reply by Zalorpas** [reply]
This is my understanding. I attach you one RAW file as an example.
[exzisus00.0.zip](https://github.com/keirf/greaseweazle/files/15448216/exzisus00.0.zip)

**Reply by keirf** [reply]
I had a look. The N field is a complete lie. The data area is 1024 bytes and CRC covers all those bytes. I'll see what I can do.
